tests/ts_http_api_SUITE.erl

```diff
@@ -0,0 +1,368 @@
+
+
+post_several_rows_test(Cfg) ->
+    RowStrs = string:join([row("q1", "w2", 20, 150), row("q1", "w1", 20, 119)],
+                          ", "),
+    Body = io_lib:format("[~s]", [RowStrs]),
+    {ok, "200", Headers, RespBody} = post_data("bob", Body, Cfg),
+    "application/json" = content_type(Headers),
+    RespBody = success_body().
+
+post_row_to_nonexisting_table_test(Cfg) ->
+    RowStr = row("q1", "w1", 30, 142),
+    {ok,"404", Headers, Body} = post_data("bill", RowStr, Cfg),
+    "text/plain" = content_type(Headers),
+    "Table \"bill\" does not exist" = Body.
+
+%%% list_keys
+list_keys_test(Cfg) ->
+    {"200", Headers, Body} = list_keys("bob", Cfg),
+    "text/plain" = content_type(Headers),
+    RecordURLs = string:tokens(Body, "\n"),
+    %% do a get on each key
+    lists:foreach(
+      fun(URL) ->
+              io:format("K ~p\n", [URL]),
+              {ok, "200", _Headers, _Body} = ibrowse:send_req(URL, [], get)
+      end,
+      RecordURLs).
+
+
+list_keys_nonexisting_table_test(Cfg) ->
+    {"404", Headers, Body} = list_keys("john", Cfg),
+    "text/plain" = content_type(Headers),
+    "Table \"john\" does not exist" = Body.
+
+%%% select
+select_test(Cfg) ->
+    Select = "select * from bob where a='q1' and b='w1' and c>1 and c<99",
+    {ok,"200", Headers, Body} = execute_query(Select, Cfg),
+    "application/json" = content_type(Headers),
+    io:format("Body ~p\n", [Body]),
+    "{\"columns\":[\"a\",\"b\",\"c\",\"d\"],"
+        "\"rows\":[[\"q1\",\"w1\",11,110],"
+        "[\"q1\",\"w1\",20,119]]}" = Body.
+
+select_subset_test(Cfg) ->
+    Select = "select * from bob where a='q1' and b='w1' and c>1 and c<15",
+    {ok, "200", Headers, Body} = execute_query(Select, Cfg),
+    "application/json" = content_type(Headers),
+    "{\"columns\":[\"a\",\"b\",\"c\",\"d\"],"
+        "\"rows\":[[\"q1\",\"w1\",11,110]]}" = Body.
+
+invalid_select_test(Cfg) ->
+    Select = "select * from bob where a='q1' and c>1 and c<15",
+    %% @todo: this really ought to be a 4XX error, but digging into the errors
+    %% from riak_ql might be too much for this API.
+    {ok, "500", Headers, Body} = execute_query(Select, Cfg),
+    "text/plain" = content_type(Headers),
+    "Execution of select query failed on table \"bob\" (The 'b' parameter is part the primary key but not specified in the where clause.)"
+        = Body.
+
+invalid_query_test(Cfg) ->
+    Select = "OHNOES A DANGLING QUOTE ' ",
+    {ok, "400", Headers, Body} = execute_query(Select, Cfg),
+    "text/plain" = content_type(Headers),
+    "Query error: Unexpected token '''." = Body.
+
+%%% delete
+delete_data_existing_row_test(Cfg) ->
+    {ok, "200", Headers, Body} = delete("bob", "q1", "w1", 11, Cfg),
+    "application/json" = content_type(Headers),
+    Body = success_body(),
+    Select = "select * from bob where a='q1' and b='w1' and c>1 and c<99",
+    {ok, "200", _Headers2,
+     "{\"columns\":[\"a\",\"b\",\"c\",\"d\"],\"rows\":[[\"q1\",\"w1\",20,119]]}"} =
```

Comment:
A more flexible JSON parser-based test might be less brittle than an exact string match here